jib-core/src/main/java/com/google/cloud/tools/jib/filesystem/LockFile.java

```diff
@@ -62,15 +62,8 @@ public static LockFile lock(Path lockFile) throws IOException {
 
     Files.createDirectories(lockFile.getParent());
     FileOutputStream outputStream = new FileOutputStream(lockFile.toFile());
-    FileLock fileLock = null;
-    try {
-      fileLock = outputStream.getChannel().lock();
+    try (FileLock fileLock = outputStream.getChannel().lock()) {
       return new LockFile(lockFile, fileLock, outputStream);
```

Comment:
To me, this change seems to make a difference on whether the `fileLock` is closed or not

- Prev: fileLock object is not close upon returning LockFile instance.
- Now: fileLock object is closed upon returning LockFile instance.

Reply:
@suztomo is right. This static `lock()` method is a "constructor": to instantiate and returns a working `LockFile` object. It is the caller that should close the instance.

Reply:
It is interesting that this issue showed up [again](https://github.com/GoogleContainerTools/jib/pull/3367). Let's mark this as a false positive?

Reply:
There are actually two `AutoCloseable` classes. In #3367, it was the `OutputStream`. Here, it's `FileLock`.

Reply:
oh good eye! I missed this.

Reply:
Ah, I missed the other related PR. Will make a note in sonar and close this one too.